src/Service/ElasticSearchService.php

```diff
@@ -0,0 +1,246 @@
+<?php
+
+/**
+ * @file
+ * Contains service to integrate with ElasticSearch.
+ */
+
+namespace App\Service;
+
+use Psr\Cache\InvalidArgumentException;
+use Symfony\Component\Cache\Adapter\FilesystemAdapter;
+use Symfony\Contracts\HttpClient\Exception\ClientExceptionInterface;
+use Symfony\Contracts\HttpClient\Exception\RedirectionExceptionInterface;
+use Symfony\Contracts\HttpClient\Exception\ServerExceptionInterface;
+use Symfony\Contracts\HttpClient\Exception\TransportExceptionInterface;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+
+/**
+ * Class ElasticSearchService.
+ */
+class ElasticSearchService implements SearchServiceInterface
+{
+    private const SEARCH_SIZE = 100;
+    private const CACHE_ID = 'ScrollId_cache';
+    private const SCROLL_ID_TTL = 60;
+
+    private $elasticSearchURL;
+
+    /** @var HttpClientInterface $httpClient */
+    private $httpClient;
+
+    /** @var FilesystemAdapter $cache */
+    private $cache;
+
+    /**
+     * StatisticsExtractionService constructor.
+     *
+     * @param HttpClientInterface $httpClient
+     *   The http client
+     * @param string $boundElasticSearchURL
+     *   URL of ElasticSearch instance
+     */
+    public function __construct(HttpClientInterface $httpClient, string $boundElasticSearchURL)
+    {
+        $this->elasticSearchURL = $boundElasticSearchURL;
+        $this->httpClient = $httpClient;
+
+        $this->cache = new FilesystemAdapter();
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * @suppress PhanTypeInvalidThrowsIsInterface
+     *
+     * @throws ClientExceptionInterface
+     * @throws InvalidArgumentException
+     * @throws RedirectionExceptionInterface
+     * @throws ServerExceptionInterface
+     * @throws TransportExceptionInterface
+     */
+    public function getLogsFromSearch(\DateTime $date, string $message): array
+    {
+        $index = 'stats_'.$date->format('d-m-Y');
+        if (!$this->indexExists($index)) {
+            // The empty array is the same as no more results for this date. So an non existing index is handled in the
+            // same way as no more data for that date.
+            return [];
+        }
+
+        // Build and send request to elastic-search.
+        $jsonQuery = $this->buildRequestBody($message);
+        $response = $this->httpClient->request('POST', $this->elasticSearchURL.$this->buildSearchPath($index), [
+            'body' => $jsonQuery,
+            'headers' => [
+                'Content-Type: application/json',
+                'Content-Length: '.strlen($jsonQuery),
+            ],
+        ]);
+        $results = json_decode($response->getContent());
+
+        // Save the current scroll id (it might change from request to request).
+        $this->getScrollId($results->_scroll_id);
```

Comment:
does not look like a save function

Reply:
Added new set and get functions to make it more clear.